.ci/test.sh

```diff
@@ -110,6 +110,9 @@ if [[ $TASK == "lint" ]]; then
         'r-lintr>=3.1.2'
     # shellcheck disable=SC1091
     source activate "${CONDA_ENV}"
+    pwsh -command "Install-Module -Name PSScriptAnalyzer -Scope CurrentUser -SkipPublisherCheck"
+    echo "Linting PowerShell code"
+    pwsh -file "./.ci/lint-powershell.ps1" || exit 0
```

Comment:
Because this doesn't require anything installed with `conda` (we're relying on the `pwsh` that comes pre-installed on the runner), I think we should move this up to before `conda create`. Two reasons:

1. prevents the case where we wait for `conda install` to complete (which can sometimes take a few minutes) and then find out that the job is going to fail because of issues found by `PSScriptAnalyzer`.
2. prevents where `source activate "${CONDA ENV}"` modifies the environment in a way that causes issues for `pwsh` (e.g. changes `PATH` / `LD_LIBRARY_PATH` and causes an incompatible mix of libraries to be loaded)

Reply:
Good catch! Totally makes sense. Will push changes right now.